src/Controller/HomeController.php

```diff
@@ -11,7 +11,7 @@ class HomeController
     public function home()
     {
         $content = Trending::fetch($this->getArgs());
-        $offset = $_GET['offset'] ?? App::get('config')['offset'];
+        $offset = $_GET['offset'] ?? (new App)->registry['config']['offset'];
```

Comment:
I understand why you need to do this but this approach won't pass my code review 🙅‍♂ 

At this point in the execution flow, you have 2 instances of `App`, it is expensive. Every instantiation tries to set error reporting and load configurations for no reason (it's already done from `index.php). Moreover, in the issue description, I've explicitly asked to have only one instance (point 2 in the Proposal section).

You have to think of an alternative way of passing configuration to controllers. Feel free to ask me for suggestions if you find yourself stuck.

Reply:
@vladimino see my try in last commit. not sure if that's good at all .. seems unreadable to me :D